File: Data Structures & Algorithms/longest-consecutive-sequence/submission-20.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        num_set = set(nums)  
        res = 0

        for num in nums: 
            if num - 1 in num_set: 
                continue 

            count = 1 
            cur = num 

            while cur + 1 in num_set: 
                count += 1 
                cur += 1 

            res = max(res, count) 

        return res

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-20.js
class Solution {
    /**
     * @param {string} s
     * @param {number} k
     * @return {number}
     */
    characterReplacement(s, k) {
        let l = 0;
        let count = {};
        let max = 0

        for (let r = 0; r < s.length; r ++) {
            let cur = s[r];
            count[cur] = (count[cur] || 0) + 1;
            let curLen = r - l + 1

            while ((curLen - Math.max(...Object.values(count))) > k) {
                // shrink window
                console.log(`Len: ${curLen}, r:${r}, l:${l}`)
                console.log(count)
                let deleted = s[l];
                count[deleted] --
                l ++
                curLen = r - l + 1
            }

            max = Math.max(curLen, max)
        }

        return max
    }
}
